src/js/components/FocusedContainer.js

```diff
@@ -18,44 +13,51 @@ export const FocusedContainer = ({
   const [bodyOverflowStyle, setBodyOverflowStyle] = useState('');
   const ref = useRef(null);
 
+  const root = useContext(RootsContext);
+  const [updatedRoot, setUpdatedRoot] = useState(root);
+
   useEffect(() => {
-    const removeTrap = () => {
-      const child = ref.current;
-      getBodyChildElements()
-        .filter(isNotAncestorOf(child))
-        .forEach(makeNodeFocusable);
-      if (restrictScroll) {
-        document.body.style.overflow = bodyOverflowStyle;
-      }
-    };
+    // make sure value of null is not added to array
+    if (ref.current) setUpdatedRoot([...root, ref.current]);
+  }, [root]);
 
-    const handleTrapFocus = () => {
-      const child = ref.current;
-      getBodyChildElements()
-        .filter(isNotAncestorOf(child))
-        .forEach(makeNodeUnfocusable);
+  useEffect(() => {
+    if (
+      bodyOverflowStyle !== 'hidden' &&
+      !hidden &&
+      restrictScroll &&
+      trapFocus
+    ) {
+      setBodyOverflowStyle(document.body.style.overflow);
+      document.body.style.overflow = 'hidden';
+    }
 
-      if (restrictScroll && bodyOverflowStyle !== 'hidden') {
-        setBodyOverflowStyle(document.body.style.overflow);
-        document.body.style.overflow = 'hidden';
+    return () => {
+      if (restrictScroll) {
+        document.body.style.overflow = bodyOverflowStyle;
       }
     };
+  }, [bodyOverflowStyle, hidden, trapFocus, restrictScroll]);
 
+  useEffect(() => {
     const timer = setTimeout(() => {
-      if (!hidden && trapFocus) {
-        handleTrapFocus();
+      if (!hidden && trapFocus && root && root[0] !== null) {
```

Comment:
I would think `roots` would never have null entries.

Reply:
In the browser, I wasn't having any issues, but when I ran the jest tests I was getting failures. Not sure if it was specific to the jest environment but adding a check resolved it